### ISSUE TITLE
Container startup errors are muted

### ISSUE THREAD

**Issue by nicolayr** [testcontainers/testcontainers-node#87]
I recently ran into an issue where a container failed to start, and the problem was tricky to track down because `testcontainers` does not expose the error. Even when running with `DEBUG=testcontainers`, the output does not include errors at startup. In the end I had to manually start the container with `docker run <image>` in order to find the error.

We're starting the container with the following code (simplified):

```javascript
try {
  logger.info('Starting container');
  const container = await new GenericContainer('image-name')
    .withCmd(['node', 'test-script'])
    .withExposedPorts(3000)
    .withWaitStrategy(Wait.forLogMessage('Ready on'))
    .start();

  /* ... stuff ... */
} catch (err) {
  logger.error('Failed to do stuff', err);
  process.exit(2);
}
```

The output I get from this is unhelpful:
```
$ DEBUG=testcontainers node do-stuff.js
[INFO] - Starting container...
  testcontainers INFO: Using Docker defaults +0ms
  testcontainers INFO: Creating container for image: image-name:latest +48ms
  testcontainers INFO: Starting container with ID: <hash> +53ms
  testcontainers DEBUG: Waiting for container to be ready +292ms
  testcontainers DEBUG: Waiting for log message "Ready on" +1ms
[ERROR] - Failed to do stuff
```

When I start the container manually with `docker run image-name` I get the actual error message (in my case, a simple syntax error in `test-script.js`).

Ideally the error would be exposed in the catch block in the code above. Currently `err` is `undefined`.

**Comment by cristianrgreco** [comment]
Duplicate of https://github.com/testcontainers/testcontainers-node/issues/89

**Comment by nicolayr** [comment]
Thanks for looking at this @cristianrgreco! I now get useful logs when running with `DEBUG=testcontainers`.

It still doesn't appear to be possible to get the container logs from the code, though. Or am I missing something?

**Comment by cristianrgreco** [comment]
@nicolayr Yes that's right, at the moment you can only get logs from a started container, so if the container fails to start you won't be able to get the logs from the code, but now you can see them via `DEBUG=testcontainers`. 

Once the container has started successfully you can do:

```javascript
(await container.logs())
    .on('data', data => ...)
    .on('err', err => ...)
```

Is there a reason you'd want to programmatically get the logs for a container that's failed to start?

**Comment by nicolayr** [comment]
Yes, the reason is that we're using testcontainers to run automatic tests in travis. If the container fails on creation we don't get any useful information in the logs, which makes it difficult to determine what went wrong.

However I was able to work around this by setting `DEBUG` programmatically, with the gotcha that it has to be set before the testcontainers module is imported/required, eg:

```javascript
const process = require('process');
process.env.DEBUG = 'testcontainers'; // Has to happen before require('testcontainers');
const { GenericContainer, Wait } = require('testcontainers');
```

**Comment by cristianrgreco** [comment]
@nicolayr I see. Is there any reason you don't add the environment variable to your `.travis.yml`?
```
env:
  - DEBUG=testcontainers
```
https://docs.travis-ci.com/user/environment-variables/

**Comment by nicolayr** [comment]
That would probably do the job as well. In our case we have a fairly complex build using Gradle, so adjusting things in travis would just add one more thing you have to check to understand how the build works.

In any case I just mentioned it in case it might help someone else in the future :)
